index.js

```diff
@@ -82,6 +82,32 @@ class Queue extends EventEmitter {
 
   async stop() {
     this.exiting = true;
+    this.paused = true;
+    // get all outstanding and waiting jobs and cancel them:
+    const outstanding = await this.findJobs({
+      $or: [
+        { status: 'processing' },
+        { status: 'waiting' }
+      ]
+    });
+    if (this.processingStatuses) {
+      outstanding.forEach(p => {
+        this.processingStatuses[p.status].dec({ jobName: p.name }, 1);
+        this.processingStatuses.cancelled.inc({ jobName: p.name }, 1);
+      });
+    }
+    await this.db.update({
```

Comment:
we should create a cancel function that takes an id and then emits the cancel event.  for this, we'd findJobs that are processing and then call .cancel for all of them

Reply:
the cancel method should handle prom too